client/src/routes/carbonsensor/carbon.js

```diff
@@ -0,0 +1,255 @@
+      return (
+        <div>
+          <Loader />
+        </div>
+      );
+    }
+    return (
+      <div className="container-fluid ">
+        <div className="row">
+          <div className="col-sm-12">
+            {' '}
+            <h3>{info.Name}</h3>
+            <h5>{info.Type} (ppm)</h5>
+            <span>Port: {this.parseStatus(info.Status)}</span>
+            <hr />
+          </div>
+        </div>
+
+        <div className="row mb-9">
+          <div className="col-sm-7">
+            <div className={style.lineChart}>
+              <Line data={getChartData(events)} options={options} />
+            </div>
+            <div className="col-sm-12">
+              <div style={sliderStyle}>
+                <b>
+                  <Slider
+                    dots
+                    step={1}
+                    onChange={this.handleOnSliderChange}
+                    marks={marks}
+                    max={4}
+                    defaultValue={0}
+                    dotStyle={sliderOption}
+                  />
+                </b>
+              </div>
+              <br />
+              <hr />
+            </div>
+          </div>
+
+          <div className="col-sm-5">
+            <Icon text={this.text} />
+            {/* <h1>{events.slice(-1)[0].signal}</h1> */}
+            <button
+              type="button"
+              className="btn btn-outline-primary"
+              checked={this.parseStatus(info.Status)}
+              onClick={this.statusOnClick(info.Status)}
+            >
+              {this.text}
+            </button>
+          </div>
+        </div>
+
+        <div className="col-sm-11">
+          <BootstrapTable
+            data={events}
+            pagination
+            options={options}
+            hover
+            version="4"
+          >
+            <TableHeaderColumn dataField="EventID" dataSort isKey>
+              ID
+            </TableHeaderColumn>
+            <TableHeaderColumn dataField="name">Device Name</TableHeaderColumn>
+            <TableHeaderColumn dataField="device_type">
+              Device Type
+            </TableHeaderColumn>
+            <TableHeaderColumn
+              dataField="CreatedAt"
+              dataFormat={this.setDateFormat}
+            >
+              Event created
+            </TableHeaderColumn>
+            <TableHeaderColumn dataField="signal">Signal</TableHeaderColumn>
+          </BootstrapTable>
+        </div>
+      </div>
+    );
+  }
+
+  setDateFormat = date => {
+    return moment(date).format('YYYY-MM-DD HH:mm:ss');
+  };
+
+  getFilterData = (filter, period) => {
+    const { dispatchSetFilter } = this.props;
+    dispatchSetFilter({
+      from: moment().subtract(filter, period),
+      value: moment().subtract('hours', 2),
+    });
+  };
+
+  statusOnClick(status) {
+    return () => this.handleOnClick(status);
+  }
+
+  parseStatus = status => {
+    switch (status) {
+      case 0:
+        this.text = 'OFF';
```

Comment:
This is not reasonable, better calculate this text in a separate method basing on props

Reply:
I think the whole method is not reasonable